meson.build

```diff
@@ -0,0 +1,231 @@
+project('gst-shark', 'c',
+  version : '0.7.5.1',
+  meson_version : '>= 0.58',)
+
+# Project version
+gst_shark_version = meson.project_version()
+version_arr = gst_shark_version.split('.')
+gst_shark_version_major = version_arr[0].to_int()
+gst_shark_version_minor = version_arr[1].to_int()
+gst_shark_version_micro = version_arr[2].to_int()
+gst_shark_version_nano = version_arr[3].to_int()
+
+gst_shark_version_short = f'@gst_shark_version_major@.@gst_shark_version_minor@.@gst_shark_version_micro@'
+
+# Create an empty configuration object to set config.h information
+cdata = configuration_data()
+
+# Get an object returns describing a compiler
+cc = meson.get_compiler('c')
+
+# Define header directories
+gst_shark_inc_dir = include_directories('plugins/tracers')
+
+# Define compiler args and include directories
+gst_c_args = ['-DHAVE_CONFIG_H']
+configinc = include_directories('.')
+
+# Find external dependencies
+glib_dep      = dependency('glib-2.0',           version : '>=2.44.0', required : true)
+gst_dep       = dependency('gstreamer-1.0',      version : '>=1.8.1', required : true)
+gst_base_dep  = dependency('gstreamer-base-1.0', version : '>=1.0.0', required : true)
+xmllib_dep    = dependency('libxml-2.0',         version : '>=1.0.0', required : true)
+gio_dep       = dependency('gio-2.0',            version : '>=1.0.0', required : true)
+gst_check_dep = dependency('gstreamer-check-1.0',version : '>=1.0.5', required : false)
+gvc_dep       = dependency('libgvc',             version : '>=2.38', required : false)
+
+## Dependencies
+# Define gst shark lib main dependencies
+gst_shark_deps = [glib_dep, xmllib_dep, gst_dep, gio_dep]
+# Define gst shark tracer main dependencies
+gst_shark_tracers_deps = [glib_dep, xmllib_dep, gst_dep]
+# Define test main dependencies
+test_gst_shark_deps = [gst_base_dep, gst_check_dep, gst_dep, glib_dep]
+
+# Graphviz enablement
+if gvc_dep.found() and get_option('enable-graphviz')
+  cdata.set('ENABLE_GRAPHVIZ', 1, description: 'Define this to enable Graphviz capabilities')
+  gst_shark_deps += [gvc_dep]
+  test_gst_shark_deps += [gvc_dep]
+  message('Graphviz enabled')
+else
+  warning('Graphviz disabled, if required, install it and enable it with -Denable-graphviz=true')
+endif
+
+# Define gstreamer API version
+apiversion = '1.0'
+
+# Define installation directories
+prefix = get_option('prefix')
+lib_install_dir = join_paths(get_option('prefix'), get_option('libdir'))
```

Comment:
Is this OK? A.f.a.i.k, libdir has the prefix already integrated.

Reply:
It does not.

However when passed as an argument to `install_dir: lib_install_dir`, you do not need a prefix, since relative paths will always be installed relative to the prefix.

This doesn't work for plugins_install_dir, one line down, because it is configured into a `cdata.set_quoted('PLUGINDIR', ...)`.

Reply:
Ok so I will leave it how it is now